packages/test/test-utils/src/localLoader.ts

```diff
@@ -45,13 +50,40 @@ export function createLoader(
 	});
 }
 
+/**
+ * Creates a loader with the given package entries and driver.
+ * @param packageEntries - A list of code details to Fluid entry points.
+ * @param documentServiceFactory - the driver factory to use
+ * @param urlResolver - the url resolver to use
+ * @param options - loader options
+ * @internal
+ */
+export function createLoaderProps(
+	packageEntries: Iterable<[IFluidCodeDetails, fluidEntryPoint]>,
+	documentServiceFactory: IDocumentServiceFactory,
+	urlResolver: IUrlResolver,
+	logger?: ITelemetryBaseLogger,
+	options?: ILoaderOptions,
+): ILoaderProps {
+	const codeLoader: ICodeDetailsLoader = new LocalCodeLoader(packageEntries);
+
+	return {
+		urlResolver,
+		documentServiceFactory,
+		codeLoader,
+		logger,
+		options,
+	};
+}
+
 /**
  * Creates a detached Container and attaches it.
  * @param source - The code details used to create the Container.
  * @param loader - The loader to use to initialize the container.
  * @param attachRequest - The request to create new from.
  * @legacy
  * @alpha
+ * @deprecated - use createAndAttachContainerUsingProps function instead
```

Comment:
Yeah, nothing should get marked deprecated in this PR since there's not a ready replacement - the deprecation should come simultaneously or after promoting these new functions to alpha+legacy.

Reply:
Just alpha is fine, right? Or until they become legacy, we can't deprecate?

Reply:
And if we don't deprecate, why would the partners app try anything else? We are providing the alternative. If they are not ready, the deprecated ones are still available and we will not remove those, until these apis are not finalized.

Reply:
Customers must not use `@alpha` in a shipping app, this is for iterating on design and getting feedback.

`@alpha` + `@legacy` is special, and permitted in shipping apps.

So, customers should not use these new APIs in production until they are promoted to `@alpha` + `@legacy`, and in the meantime we need to retain a non-deprecated `@alpha` + `@legacy` option.

Reply:
I tried to add these separate apis: https://github.com/microsoft/FluidFramework/pull/22902/commits/e161f42e3fdbc077b1a2350fadbd16363b22ad8d

Please take a look and let me know what you think. For now, I haven't modified the rest of PR. But just wanted to get review on APIs. before I work on the rest of PR.